crates/vm/levm/src/gas_cost.rs

```diff
@@ -180,27 +180,31 @@ pub const MODEXP_STATIC_COST: u64 = 0;
 pub const MODEXP_DYNAMIC_BASE: u64 = 200;
 pub const MODEXP_DYNAMIC_QUOTIENT: u64 = 3;
 
-pub fn exp(exponent: U256) -> Result<U256, OutOfGasError> {
+pub fn exp(exponent: U256) -> Result<u64, VMError> {
     let exponent_byte_size = (exponent
         .bits()
         .checked_add(7)
-        .ok_or(OutOfGasError::GasCostOverflow)?)
+        .ok_or(VMError::OutOfGas(OutOfGasError::GasCostOverflow))?)
```

Comment:
It wasn't neceesary to add `VMError::OutOfGas`. `thiserror` crate handles that. Doesn't matter much though, just a thing i noticed